code/game/objects/items/stunbaton.dm

```diff
@@ -173,9 +172,32 @@
 		var/mob/living/carbon/human/H = L
 		H.forcesay(GLOB.hit_appends)
 
-
 	return 1
 
+/// After a target is hit, we do a chunk of stamina damage, along with other effects.
+/// After a period of time, we then check to see what stun duration we give.
+/obj/item/melee/baton/proc/stun_effect(mob/living/target)
+	target.Jitter(20)
+	target.confused = 10

```

Comment:
make this set the victim's confusion value to either 10 or their current confusion value, whichever is higher (or just add 10 to the victim's confusion value).

Reply:
Adding 10 has the issue of just stacking it up constantly each hit which doesn’t feel quite right in game, though going but whatever is highest is👍

Reply:
target.confused = max(10, target.confused)